bpf/lib/nodeport.h

```diff
@@ -1101,6 +1123,35 @@ static __always_inline bool snat_v4_needed(struct __ctx_buff *ctx, __be32 *addr,
 	 */
 	return false;
 # endif
+
+#if defined(ENABLE_EGRESS_GATEWAY)
+	/* If the packet is destined to an entity inside the cluster, either EP
+	 * or node, skip SNAT since only traffic leaving the cluster is supposed
+	 * to be masqueraded with an egress IP.
+	 */
```

Comment:
It took me a while to realize (and it's obvious in retrospect) but what this patch does is to move the egress gw check _before_ the `IPV4_SNAT_EXCLUSION_DST_CIDR` check.

How about we add a comment here explaining why the egress gw check needs to happen before the `IPV4_SNAT_EXCLUSION_DST_CIDR`?

If we do so, maybe also change the commit message from:
> To fix this, this commit changes the snat_v4_needed() logic to match the
one we use in handle_ipv4_from_lxc().

To something like:
> To fix this, this commit changes the snat_v4_needed() logic to match the
one we use in handle_ipv4_from_lxc() and executes it before the  `IPV4_SNAT_EXCLUSION_DST_CIDR` check.

Reply:
Done :heavy_check_mark: